builder/resubmitter.go

```diff
@@ -0,0 +1,42 @@
+package builder
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+type Resubmitter struct {
+	mu     sync.Mutex
+	cancel context.CancelFunc
+}
+
+func (r *Resubmitter) newTask(repeatFor time.Duration, interval time.Duration, fn func() error) error {
+	repeatUntilCh := time.After(repeatFor)
+
+	r.mu.Lock()
+	if r.cancel != nil {
+		r.cancel()
+	}
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
cancel should happen after repeatFor, not after exiting from the function!

Reply:
We kind of want to return the first return value, otherwise it'd work
Not returning the first value makes both testing and debugging a bit more difficult, since you don't see any issues in the responses to the BN